lisp/ess-developer.el

```diff
@@ -377,21 +430,81 @@ If ALL is non-nil, deactivate in all open R buffers."
+                   (ess-developer--get-package-path)))
+         (name (when path
+                 (file-name-nondirectory (directory-file-name path)))))
+    ;; Ask package directory only when not obvious
+    (unless (or ess-developer-local-package path)
+      (setq path (read-directory-name "Package: " path nil t nil)))
+    (unless (file-exists-p (expand-file-name ess-developer-root-file path))
       (error "Not a valid package. No '%s' found in `%s'."
-             ess-developer-root-file package))
-    (message "Loading %s" (abbreviate-file-name package))
-    (ess-eval-linewise
-     (format ess-developer-load-package-command package))))
+             ess-developer-root-file path))
+    (message msg name)
+    (let ((name (or (car ess-developer-local-package)
+                    (ess-developer--get-package-name path))))
+      (ess-developer--init-process-local-vars name path))
+    (ess-eval-linewise (format command path))))
+
+(defun ess-developer--init-process-local-vars (name path)
+  (with-ess-process-buffer nil
+    (setq-local ess-developer-local-package (cons name path))))
+
+(defun ess-developer-load-package ()
+  "Interface to load_all() function from devtools package.
+
+Without prefix, load the package. With single prefix, recompile
+before loading. With double prefix, unload the package."
+  (interactive)
+  (cond ((equal current-prefix-arg '(16))
+         (ess-developer-unload-package))
+        ((equal current-prefix-arg '(4))
+         (ess-developer-send-process "devtools::load_all('%s', recompile = TRUE)\n"
+                                     "Recompiling %s"))
+        (t
+         (ess-developer-send-process "devtools::load_all('%s')\n"
+                                     "Loading %s"))))
+
+(defun ess-developer-unload-package ()
+  "Interface to unload() function from devtools package."
+  (interactive)
+  (ess-developer-send-process "devtools::unload('%s')\n"
+                              "Unloading %s"))
+
+(defun ess-developer-check-package ()
+  "Interface to checking functions from devtools package.
+
+Without prefix, run the unit tests. With single prefix, perform a
+R CMD check. With double prefix, check the reverse dependencies."
+  (interactive)
+  (cond ((equal current-prefix-arg '(16))
+         (ess-developer-send-process "devtools::revdep_check('%s')\n"
+                                     "Checking reverse dependencies of %s"))
+        ((equal current-prefix-arg '(4))
+         (ess-developer-send-process "devtools::check('%s')\n"
+                                     "Checking %s"))
+        (t
+         (ess-developer-send-process "devtools::test('%s')\n"
+                                     "Testing %s"))))
+
+(defun ess-developer-document-package ()
+  "Interface to document() from devtools package."
+  (interactive)
+  (ess-developer-send-process "devtools::document('%s')\n"
+                              "Documenting %s"))
+
+(defun ess-developer-install-package ()
```

Comment:
You don't need to kill the buffer. Just call `ess-quit` and start the package with R. I am doing this manualy all the time but never got down to implementing it :(